sos/plugins/extras.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2019 Red Hat, Inc., Pavel Moravec <pmoravec@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, UbuntuPlugin, DebianPlugin
+import os
+
+
+class Extras(Plugin, RedHatPlugin, UbuntuPlugin, DebianPlugin):
+    """Extras plugin to collect user declared commands outputs
+       The plugin traverses /etc/sos.extras.d directory and for each
+       file there, it executes each line as a bash command.
```

Comment:
It doesn't ... it just passes it straight to `add_cmd_output()` which very explicitly disables execution under a shell. We do not want to ever imply that we support shell style notation here - it's way too unpredictable.